File: ruby/lib/woothee/appliance.rb
# -*- coding: utf-8 -*-

require 'woothee/dataset'
require 'woothee/util'

module Woothee::Appliance
  extend Woothee::Util

  def self.challenge_playstation(ua, result)
    data = case
           when ua.index('PSP (PlayStation Portable);') then Woothee::DataSet.get('PSP')
           when ua.index('PlayStation Vita') then Woothee::DataSet.get('PSVita')
           when ua.index('PLAYSTATION 3 ') || ua.index('PLAYSTATION 3;') then Woothee::DataSet.get('PS3')
           else nil
           end
    return false unless data

    update_map(result, data)
    true
  end

  def self.challenge_nintendo(ua, result)
    data = case
           when ua.index('Nintendo 3DS;') then Woothee::DataSet.get('Nintendo3DS')
           when ua.index('Nintendo DSi;') then Woothee::DataSet.get('NintendoDSi')
           when ua.index('Nintendo Wii;') then Woothee::DataSet.get('NintendoWii')
           when ua.index('(Nintendo WiiU)') then Woothee::DataSet.get('NintendoWiiU')
           else nil
           end
    return false unless data

    update_map(result, data)
    true
  end

  def self.challenge_digitaltv(ua, result)
    data = if ua.index('InettvBrowser/')
             Woothee::DataSet.get('DigitalTV')
           else
             nil
           end
    return false unless data

    update_map(result, data)
    true
  end
end

File: python/lib/woothee/appliance.py
# -*- coding: utf-8 -*-

from . import dataset
from . import util

def challenge_playstation(ua, result):
  if 'PSP (PlayStation Portable);' in ua:
    data = dataset.get('PSP')
  elif 'PlayStation Vita' in ua:
    data = dataset.get('PSVita')
  elif 'PLAYSTATION 3 ' in ua or 'PLAYSTATION 3;' in ua:
    data = dataset.get('PS3')
  else:
    return False
  util.update_map(result, data)
  return True

def challenge_nintendo(ua, result):
  if 'Nintendo 3DS;' in ua:
    data = dataset.get('Nintendo3DS')
  elif 'Nintendo DSi;' in ua:
    data = dataset.get('NintendoDSi')
  elif 'Nintendo Wii;' in ua:
    data = dataset.get('NintendoWii')
  elif '(Nintendo WiiU)' in ua:
    data = dataset.get('NintendoWiiU')
  else:
    return False
  util.update_map(result, data)
  return True

def challenge_digitaltv(ua, result):
  if 'InettvBrowser/' in ua:
    data = dataset.get('DigitalTV')
  else:
    return False
  util.update_map(result, data)
  return True
